storage/oasis/nodeapi/file/node.go

```diff
@@ -0,0 +1,293 @@
+package file
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"encoding/json"
+	"fmt"
+
+	"github.com/akrylysov/pogreb"
+	beacon "github.com/oasisprotocol/oasis-core/go/beacon/api"
+	coreCommon "github.com/oasisprotocol/oasis-core/go/common"
+	consensus "github.com/oasisprotocol/oasis-core/go/consensus/api"
+	genesis "github.com/oasisprotocol/oasis-core/go/genesis/api"
+
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis/nodeapi"
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis/nodeapi/damask"
+)
+
+// FileConsensusApiLite provides access to the consensus API of an Oasis node.
+// Since FileConsensusApiLite is backed by a file containing the cached responses
+// to `ConsensusApiLite` calls, this data is inherently compatible with the
+// current indexer and can thus handle heights from both Cobalt/Damask.
+type FileConsensusApiLite struct {
+	db        pogreb.DB
+	damaskApi *damask.DamaskConsensusApiLite
+}
+
+type ConsensusApiMethod func() (interface{}, error)
+
+var _ nodeapi.ConsensusApiLite = (*FileConsensusApiLite)(nil)
+
+func NewFileConsensusApiLite(filename string, client *consensus.ClientBackend) (*FileConsensusApiLite, error) {
+	db, err := pogreb.Open(filename, &pogreb.Options{BackgroundSyncInterval: -1})
+	if err != nil {
+		return nil, err
+	}
+	var damaskApi *damask.DamaskConsensusApiLite
+	if client != nil {
+		damaskApi = damask.NewDamaskConsensusApiLite(*client)
+	}
+	return &FileConsensusApiLite{
+		db:        *db,
+		damaskApi: damaskApi,
+	}, nil
+}
+
+func generateCacheKey(methodName string, params ...interface{}) ([]byte, error) {
+	var buf bytes.Buffer
+	enc := gob.NewEncoder(&buf)
+	err := enc.Encode(methodName)
+	if err != nil {
+		return nil, err
```

Comment:
WDYT about just panicking here and below, and simplify the return value to not include `error`? It would simplify callsites, and I don't see a reason `.Encode()` to suddenly return an error anyway. Only in some corner cases like null pointers or weird types, but those properties should not change for us from height to height.

Reply:
Hm I want to test that this assumption is still true after adding RuntimeApiLite support. In particular, the `EVMSimulateCall(ctx context.Context, round uint64, gasPrice []byte, gasLimit uint64, caller []byte, address []byte, value []byte, data []byte) ([]byte, error)` seems like it could be potentially problematic.

That being said; there's no real point in error handling/retrying if this fails so I'll update it to panic. Thanks!

Reply:
:+1: 
Yeah I wasn't trying to say these won't ever fail (especially as we add/change types), just that it is very unlikely to start failing at some random block. If it fails, it will do so immediately, and we'll catch it in local testing or e2e, where panicking is fine.